src/chunks.mjs

```diff
@@ -112,37 +113,38 @@ export function listen(options) {
   };
 }
 
-
 /**
 * Prefetch a given URL with an optional preferred fetch priority
 * @param {String} url - the URL to fetch
 * @param {Boolean} [isPriority] - if is "high" priority
-* @param {Object} [conn] - navigator.connection (internal)
 * @return {Object} a Promise
 */
-export function prefetch(url, isPriority, conn) {
-  if (conn = navigator.connection) {
+export function prefetch(url, isPriority) {
+  const {connection} = navigator;
+
+  if (connection) {
     // Don't prefetch if using 2G or if Save-Data is enabled.
-    if (conn.saveData) {
+    if (connection.saveData) {
       return Promise.reject(new Error('Cannot prefetch, Save-Data is enabled'));
     }
-    if (/2g/.test(conn.effectiveType)) {
+
+    if (/2g/.test(connection.effectiveType)) {
       return Promise.reject(new Error('Cannot prefetch, network conditions are poor'));
     }
   }
 
   // Dev must supply own catch()
   return Promise.all(
       [].concat(url).map(str => {
-        if (!toPrefetch.has(str)) {
+        if (toPrefetch.has(str)) return [];
```

Comment:
Unsure if this is better to be `undefined`?

Reply:
I think that it should be fine to return `[]` as you currently have here